tabcmd/commands/server.py

```diff
@@ -48,28 +50,36 @@ def find_user_id(logger, server, username):
             Errors.exit_with_error(logger, exception=e)
 
     @staticmethod
-    def get_items_by_name(logger, item_endpoint, item_name, container=None):
+    def get_items_by_name(logger, item_endpoint, item_name: str, container: TSC.ProjectItem = None):
         item_type = type(item_endpoint).__name__
-        item_log_name = item_name
+        item_log_name: str = "[" + item_type + "] " + item_name
         if container:
-            item_log_name = container + "/" + item_log_name
-        item_log_name = "[" + item_type + "] " + item_log_name
+            item_log_name = str(container) + "/" + item_log_name
         logger.debug(_("export.status").format(item_log_name))
         req_option = TSC.RequestOptions()
         req_option.filter.add(TSC.Filter(TSC.RequestOptions.Field.Name, TSC.RequestOptions.Operator.Equals, item_name))
-        if container:
-            logger.debug("Searching in project {}".format(container))
-            req_option.filter.add(
-                TSC.Filter(TSC.RequestOptions.Field.ParentProjectId, TSC.RequestOptions.Operator.Equals, container)
-            )
+
         all_items, pagination_item = item_endpoint.get(req_option)
         if all_items is None or all_items == []:
             raise ValueError("[" + item_type + "] " + _("errors.xmlapi.not_found"))
+        if len(all_items) == 1:
+            logger.debug("Exactly one result found")
+            result = all_items
         if len(all_items) > 1:
-            logger.debug("{}+ items of this name were found. Returning first page.".format(len(all_items)))
-            logger.debug(all_items[0].name + ", " + all_items[1].name + ", " + all_items[2].name)
+            logger.debug(
```

Comment:
localize?

Reply:
rolling over to https://github.com/tableau/tabcmd/issues/155